adr/ADR-42.md

```diff
@@ -0,0 +1,94 @@
+# NATS Client LifeCycle (-ERR Protocol Error Handling)
+
+| Metadata | Value          |
+| -------- | -------------- |
+| Date     | 2024-05-31     |
+| Author   | @aricart       |
+| Status   | Implemented    |
+| Tags     | client, server |
+
+## Context and Problem Statement
+
+Client lifecycle such as connect/reconnect/liveliness (ping/pong)/LDM behaviours
+are fairly complex in a NATS client. This ADR simply documents `-ERR` protocol
+messages that are sent to a client.
+
+The `-ERR` protocol message is an important signal for clients about things that
+are incorrect from the perspective of Permissions or Authorization.
+
+## Errors
+
+### Permission Violation
+
+`Permission Violation` means that the client tried to publish or subscribe on a
+subject for which it has no permissions. This type of error can happen or
+surface at any time, as changes to permissions intentionally or not can happen.
+
+The message will include `/(Publish|Subscription) to (\S+)/` this will indicate
```

Comment:
worth calling out that people specifically should not try to match end of line as other stuff are and maybe added later

Reply:
added